app-modules/prospect/src/Http/Controllers/Api/V1/Prospects/ListProspectsController.php

```diff
@@ -0,0 +1,152 @@
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Prospect\Http\Controllers\Api\V1\Prospects;
+
+use AdvisingApp\Prospect\Http\Resources\Api\V1\ProspectResource;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\Prospect\Sorts\ProspectSourceSort;
+use AdvisingApp\Prospect\Sorts\ProspectStatusSort;
+use Dedoc\Scramble\Attributes\Example;
+use Dedoc\Scramble\Attributes\Group;
+use Dedoc\Scramble\Attributes\QueryParameter;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Resources\Json\AnonymousResourceCollection;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Support\Str;
+use Spatie\QueryBuilder\AllowedFilter;
+use Spatie\QueryBuilder\AllowedInclude;
+use Spatie\QueryBuilder\AllowedSort;
+use Spatie\QueryBuilder\Enums\FilterOperator;
+use Spatie\QueryBuilder\QueryBuilder;
+
+class ListProspectsController
+{
+    /**
+     * @response AnonymousResourceCollection<LengthAwarePaginator<ProspectResource>>
+     */
+    #[Group('Prospects')]
+    #[QueryParameter('filter[id]', description: 'Filter the results where the prospect\'s ID contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[first_name]', description: 'Filter the results where the prospect\'s first name contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[last_name]', description: 'Filter the results where the prospect\'s last name contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[full_name]', description: 'Filter the results where the prospect\'s full name contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[preferred]', description: 'Filter the results where the prospect\'s preferred name contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[sms_opt_out]', description: 'Filter the results where the prospect\'s sms_opt_out matches the provided boolean.', type: 'boolean')]
+    #[QueryParameter('filter[email_bounce]', description: 'Filter the results where the prospect\'s email_bounce matches the provided boolean.', type: 'boolean')]
+    #[QueryParameter('filter[birthdate]', description: 'Filter the results where the prospect\'s birthdate matches the provided date.', type: 'date')]
+    #[QueryParameter('filter[status]', description: 'Filter the results where the prospect\'s status matches the provided status name.', type: 'string')]
+    #[QueryParameter('filter[source]', description: 'Filter the results where the prospect\'s source matches the provided source name.', type: 'string')]
+    #[QueryParameter('filter[hsgrad]', description: 'Filter the results where the prospect\'s high school graduation year matches the provided integer.', type: 'integer')]
```

Comment:
This description doesn't seem super accurate. Since it is a dynamic filter, can't you filter based on the date with different operators?